operational_analysis/methods/plant_analysis.py

```diff
@@ -774,15 +775,22 @@ def run_regression(self, n):
             self._mse_score[n] = mean_squared_error(reg_data[:,-1], predicted_y)
             return reg
         # Machine learning models
-        else: 
+        else:
             ml = MachineLearningSetup(self.reg_model, **self.ml_setup_kwargs)
-            ml.hyper_optimize(np.array(reg_data[:,0:-1]), reg_data[:,-1], n_iter_search = 5, report = False, cv = KFold(n_splits = 2))
-            predicted_y = ml.random_search.predict(np.array(reg_data[:,0:-1]))
+            # Memoized approach for optimized hyperparameters
+            if self._run.reanalysis_product in self.opt_model:
+                self.opt_model[(self._run.reanalysis_product)].fit(np.array(reg_data[:,0:-1]), reg_data[:,-1])
+            else: # optimize hyperparameters once for each reanalysis product  
+                ml.hyper_optimize(np.array(reg_data[:,0:-1]), reg_data[:,-1], n_iter_search = 20, report = False, cv = KFold(n_splits = 5))
+                # Store optimized hyperparameters for each reanalysis product
+                self.opt_model[(self._run.reanalysis_product)] = ml.opt_model
```

Comment:
Do we also need to run line 782 again, here? I don't see any call to opt_model.fit(). Furthermore, I don't see any member MachineLearningSetup.opt_model in the source. Where is that variable coming from, and when is it fit if line 782 has not yet been run?

Reply:
Hi Jordan,  MachineLearningSetup.opt_model is assigned in machine_learning_setup.py, line 147. It is the result of the fit in line 143 there, so that's why the fit is not performed again in plant_analysis.py. On the other hand, we need to re-perform the fit in line 782 since in that case we're using a different set of input data to the regression. Hope this is clear enough, let me know if not!

Reply:
I see. I was looking at the develop branch and missed your addition to that file in this PR. This looks good. Your comments make sense and resolve the issue.